File: Java/1403. Minimum Subsequence in Non-Increasing Order.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        int s = 0;
        ArrayList<Integer> ans = new ArrayList<>();

        Arrays.sort(nums);

        for(int i: nums) s += i;
        for(int i = nums.length - 1; i >= 0; i--){
            s -= nums[i] * 2;
            ans.add(nums[i]);
            if(s < 0) break;
        }

        return ans;
    }
}

File: Python/1403. Minimum Subsequence in Non-Increasing Order.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        s = sum(nums)

        nums.sort(reverse = True)

        for i in range(len(nums)):
            s -= nums[i] * 2
            if s < 0: return nums[:i + 1]
